src/odemis/gui/comp/overlay/shapes.py

```diff
@@ -232,6 +273,12 @@ def on_char(self, evt):
                 self._shape_to_copy = self._selected_shape
                 self.cnvs.set_default_cursor(wx.CURSOR_BULLSEYE)
                 self.cnvs.request_drawing_update()
+            elif evt.GetKeyCode() == wx.WXK_CONTROL_Z:
+                self._selected_shape.undo()
+                self._selected_shape.refresh()
+            elif evt.GetKeyCode() == wx.WXK_CONTROL_Y:
+                self._selected_shape.redo()
+                self._selected_shape.refresh()
```

Comment:
That mean the undo/redo stack is per *shape*. Isn't that really odd for the user? Typically programs provide a single undo stack. Did you check with the users that it's what they want? I guess that once you're aware of it, it fulfil its need, but the fact that as soon as you select another shape you cannot undo the changes you've made by pressing Ctrl+Z is confusing.

Reply:
Right, I will change the implementation to `Typically programs provide a single undo stack.`